pretrain/validation.py

```diff
@@ -179,13 +183,26 @@ def compute_losses(
     with torch.no_grad():
         for batch in batches:
             try:
-                inputs = batch.to(device)
+                inputs = torch.tensor(batch).to(device)
                 logits = model(inputs).logits
 
                 shift_logits = logits[..., :-1, :].contiguous()
                 shift_labels = inputs[..., 1:].contiguous()
+
+                if not pack_samples:
+
+                    # If sample unpacking is used,
+                    # create a mask to indicate location of PAD tokens.
+                    # Note, PAD tokens are always set to EOS tokens,
+                    # For this reason, we want to ignore all but the
+                    # first EOS token (the real one)
+                    pad_mask = shift_labels == pad_token_id
+                    zeros = torch.zeros_like(shift_labels[...,:1])
+                    pad_mask = torch.cat((zeros, pad_mask[...,:-1]), dim=-1).bool()
+                    shift_labels[pad_mask] = -100
+
                 # Flatten the tokens
-                loss_fct = torch.nn.CrossEntropyLoss()
+                loss_fct = torch.nn.CrossEntropyLoss(ignore_index=-100) # -100 is actually the default value
```

Comment:
If it is the default value did you need to add it explicitly for some reason?

Reply:
I added it that for clarity as some might not be aware of -100 as the default value. So setting EOS labels in the lines just before to -100 is more understandable. I will remove it, and keep the comment for clarity